src/Payum/Paypal/ExpressCheckout/Nvp/ApiPermission.php

```diff
@@ -0,0 +1,76 @@
+<?php
+namespace Payum\Paypal\ExpressCheckout\Nvp;
+
+use Payum\Core\Exception\LogicException;
+use Payum\Paypal\ExpressCheckout\Nvp\Api as BaseApi;
+use Payum\Core\Exception\Http\HttpException;
+use Psr\Http\Message\RequestInterface;
+use PayPal\Auth\Oauth\AuthSignature;
+
+/**
+ * {@inheritDoc}
+ *
+ * @link https://developer.paypal.com/docs/classic/products/permissions/
+ * @link https://developer.paypal.com/docs/classic/permissions-service/integration-guide/PermissionsUsing/
+ *
+ */
+class ApiPermission extends BaseApi
+{
+
+    /**
+     * @var array
+     */
+    protected $options = array(
+        'username' => null,
+        'password' => null,
+        'signature' => null,
+        'return_url' => null,
+        'cancel_url' => null,
+        'token' => null,
+        'tokenSecret' => null,
+        'third_party_subject' => null,
+        'sandbox' => null,
+        'useraction' => null,
+        'cmd' => self::CMD_EXPRESS_CHECKOUT,
+    );
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function authorizeRequest(RequestInterface $request)
+    {
+        $request = parent::authorizeRequest($request);
+
+        $fields = array();
+        parse_str($request->getBody(), $fields);
+        $fields['SUBJECT'] = $this->options['third_party_subject'];
+        $request = $request->withBody(http_build_query($fields));
+        $authSignature = $this->generateOauthSignature($request);
+
+        return $request->withAddedHeader('X-PAYPAL-AUTHORIZATION', $authSignature);
+    }
+
+    /**
+     * Generates OAuth signature for X-PAYPAL-AUTHORIZATION header.
+     * @see https://developer.paypal.com/docs/classic/permissions-service/integration-guide/PermissionsUsing/
+     *
+     * @param RequestInterface $request
+     * $throws LogicException
+     * @return string
+     */
+    protected function generateOauthSignature(RequestInterface $request)
+    {
+        if (false == class_exists(AuthSignature::class)) {
```

Comment:
move this `if` statement to the beginning of the gateway's populateConfig method